tests/test_cases/test_package/test_package.py

```diff
@@ -41,6 +41,40 @@ async def test_stringification(dut):
     assert str(pkg2.eleven_int) == "IntegerObject(cocotb_package_pkg_2::eleven_int)"
 
 
+def get_integer(hdl):
+    if isinstance(hdl, cocotb.handle.IntegerObject):
+        return hdl.value
+    else:
+        return hdl.value.integer
+
+
+@cocotb.test()
+async def test_long_parameter(dut):
+    """
+    Test package parameter access
+
+    # On verilator:
+    # 0.00ns ERROR    VPI error
+    # 0.00ns ERROR    vl_check_format: Unsupported format (vpiIntVal) for cocotb_package_pkg_1.long_param
+    # 0.00ns ERROR    Failed to initialize test test_long_parameter
+    """
+
+    pkg1 = cocotb.packages.cocotb_package_pkg_1
+    if cocotb.SIM_NAME.lower().startswith("verilator"):
+        pass
+    else:
+        # most sims truncate the value to 32 bits, and interpret as signed despite the marking
+        assert str(pkg1.long_param) == "IntegerObject(cocotb_package_pkg_1::long_param)"
+        # should really be 0x5A89901AF1 (40 bits)
+        # -1987044623 = (two's comp) 0b10001001100100000001101011110001 = 0x5A89901AF1[31:0]
+        assert get_integer(pkg1.long_param) == -1987044623
+        assert (
+            str(pkg1.really_long_param)
+            == "IntegerObject(cocotb_package_pkg_1::really_long_param)"
+        )
+        assert get_integer(pkg1.really_long_param) == -1987044623
```

Comment:
I think the fact that this passes with the wrong value on all sims really shows the shortcomings of this. I think it makes sense to have only have vpiIntConst map to GPI_INTEGER

Reply:
I've tested all of the SV simulators except Xcelium, and none of them use `vpiIntConst` at all.
Only `vpiBinaryConst` and `vpiDecConst`.

Reply:
So is the interpretation that vpiBinaryConst = logic, and vpiDecConst = int (and longint) ? So then the size check is what we want to go with? We can discuss here https://github.com/cocotb/cocotb/pull/3691. Could we get this tc landed for now?